robolectric/src/test/java/org/robolectric/shadows/ShadowPackageManagerTest.java

```diff
@@ -211,11 +161,15 @@ public void testQueryBroadcastReceiverFailsForMissingPackageName() {
   }
 
   @Test
-  public void testQueryBroadcastReceiverFailsForMissingAction() {
+  public void testQueryBroadcastReceiver_matchAllWithoutIntentFilter() {
     Intent intent = new Intent();
     intent.setPackage(RuntimeEnvironment.application.getPackageName());
     List<ResolveInfo> receiverInfos = packageManager.queryBroadcastReceivers(intent, PackageManager.GET_INTENT_FILTERS);
-    assertTrue(receiverInfos.size() == 0);
+    assertThat(receiverInfos).hasSize(7);
+
+    for (ResolveInfo receiverInfo : receiverInfos) {
```

Comment:
Just suggestion, you could use a Custom Comparator here to do the comparison on receiverInfo object and have it compare the field values for you.   See: https://github.com/joel-costigliola/assertj-examples/blob/master/assertions-examples/src/test/java/org/assertj/examples/AssertionsWithCustomComparatorExamples.java

Reply:
Yeah, I've thought about that before, I like that it would be more readable, but for just this one instance it ends up adding more code for just one more readable assertion.

There is https://github.com/square/assertj-android and I liked the idea of that too but its deprecated as I think its considered a never ending maintenance burden

Reply:
@jongerrish I use the assertj-android and there could be a lot of automation done to it to help alleviate some of the manual maintenance burden.  There is a lot that the automated generator could generate.

http://joel-costigliola.github.io/assertj/assertj-assertions-generator.html

Might even be nice for Robolectric to autogenerate some assertions for the Shadows and custom classes.

I still would encourage using assertj-android 90% of it still works, it is just adding newer APIs just like Robolectric needs to.

Reply:
We've actually discussed generating assert subjects for truth (and/or assertj) based on annotations on the shadows so I think we're on the same page :-)